lib/Models/GtfsCatalogItem.ts

```diff
@@ -88,15 +88,29 @@ export default class GtfsCatalogItem extends AsyncMappableMixin(
   protected convertManyFeedEntitiesToBillboardData: ITransformer<
     FeedEntity[],
     VehicleData[]
-  > = createTransformer((feedEntity: FeedEntity[]) => {
-    return this.gtfsFeedEntities
-      .map((entity: FeedEntity) =>
-        this.convertFeedEntityToBillboardData(entity)
-      )
-      .filter(
-        (item: VehicleData) =>
-          item.position !== null && item.position !== undefined
-      );
+  > = createTransformer((feedEntities: FeedEntity[]) => {
+    // Sometimes the feed can contain many records for the same vehicle
+    // so we'll only display the newest record.
+    // Although technically the timestamp property is optional, if none is
+    // present we'll show the record.
+    const vehicleMap = new Map()
+    for (var i = 0; i < feedEntities.length; ++i) {
+      const entity: FeedEntity = feedEntities[i]
+      const item: VehicleData = this.convertFeedEntityToBillboardData(entity)
+
+      if (item && item.position && item.featureInfo) {
```

Comment:
I'm not a huge fan of these sorts of undefined checks but that's pure personal preference and you should ignore me if you don't agree
in general, things could evaluate to false and still be defined

Reply:
Is using the `defined` method from `"terriajs-cesium/Source/Core/defined"` more appropriate @KeyboardSounds ?

Reply:
Couldn't make typescript happy using the `defined` method :(